integration-libs/cdc/components/gigya-raas/gigya-raas.component.ts

```diff
@@ -133,4 +151,21 @@ export class GigyaRaasComponent implements OnInit {
     }
     return false;
   }
+
+  canActivate(data: GigyaRaasComponentData): Observable<boolean> {
```

Comment:
In general, it shouldn't be responsibility of components to guard a route. Instead it should be Angular route guards (which are executed earlier that any component is rendered on the anticipated page).

Othwerwise, we may potentially show a flicker of a UI of the anticipated page and then almost immediately redirect to other page.

Did you consider implementing a CMS-mapping-driven route guard instead?

Reply:
I can see the challenge is injecting a CMS component's data just for the Gigya component  🤔 

Component Data can be obtained by `uid` from `cmsService.getComponentData(uid)`

And `uid` of `GigyaRaasComponent` must be found in the whole `Page` object:
```ts
this.routingService.getNextPageContext().pipe(
  filter(isNotUndefined),
  take(1),
  switchMap((pageContext) =>
    this.cmsService.getPage(pageContext, this.shouldReload()).pipe(
      map(page: Page => /* find all (one?) component of type `GigyaRaasComponent` and pick it's `id`)
```

I'm not aware of other alternative ways to access component data in the cms-mapping driven route guard.

Pros:
- routing logic is kept in Angular Route Guards (to be precise in cms-driven Route Guards)
Cons:
- need to implement a custom logic for retrieving Gigya component ID from the anticipated `Page` object (in our custom Route Guard), just to get component's data.

What do you think about it?

Reply:
Hi Kris, yes it makes more sense. I have adapted this approach. Please check